en/peripherals/gripper.md

```diff
@@ -0,0 +1,107 @@
+
+:::note
+The [Roboclaw driver](../modules/modules_driver.md#roboclaw) isn't supported as a Gripper yet.
+However in the future it may get supported as a Gripper.
+:::
+
+## Using a Gripper
+
+For information on using a gripper in missions see [Payload Delivery Missions](../flying/package_delivery_mission.md).
+
+You can trigger the gripper manually if you've mapped `gripper open` and `gripper close` buttons in the [QGC Joystick Configuration](#qgc-joystick-configuration).
+Note that if you press **Grab** button while the gripper is opening, it will automatically abort releasing behavior and go to the Close position, effectively cancelling the release command.
+If you do this in a mission while the release is actually happening then the [delivery will be cancelled](../flying/package_delivery_mission.md#manual-control-of-gripper-in-missions).
+
+MAVLink applications, such as ground stations, can also control the gripper using the [MAV_CMD_DO_GRIPPER](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_GRIPPER) MAVLink command.
+
+
+## PX4 Configuration
+
+### Package Delivery Configuration
+
+PX4 Gripper support is tied to the package delivery feature, which must be enabled and configured in order to be able to use a gripper.
+
+1. Set [`PD_GRIPPER_EN`](../advanced_config/parameter_reference.md#PD_GRIPPER_EN) parameter to 1 (reboot required after change).
+1. Set [`PD_GRIPPER_TYPE`](../advanced_config/parameter_reference.md#PD_GRIPPER_TYPE) to match your Gripper.
+   For example, set to `Servo` for a [Servo Gripper](gripper_servo.md).
+
+### Gripper Actuator Mapping
+
+To enable the output of the Gripper, set the Function of the output port where the gripper is connected to as `Gripper` in the [Actuators](../config/actuators.md#actuator-outputs) tab.
+
+<!-- Replace this with actuator diagram -->
+
+### QGC Joystick Configuration
+
+QGroundControl [Joystick](../config/joystick.md) configuration allows you to map gripper actions to Joystick buttons, after which you will be open and close the gripper manually.
+
+Open the Joystick configuration by selecting: **QGC Logo (upper-left) > Vehicle Setup > Joystick > Button Assignment** tab.
+Select `Gripper Open` and `Gripper Close` actions for your desired joystick buttons
+
+![Gripper action mapping](../../assets/config/gripper/qgc_gripper_actions_joystick.png)
+
+You can test the actions by clicking on the mapped buttons and checking for gripper movement.
+If the gripper doesn't move as expected check the package delivery configuration and actuator mapping are set up properly.
+
+### Enable Pre-ARM Mode
+
+We need to set a parameter to enable [pre-arming](../advanced_config/prearm_arm_disarm.md).
+This keeps the motors disabled but allows the gripper to be opened and closed for attaching the payload (avoiding potential danger from spinning propellers).
+
+Set [`COM_PREARM_MODE`](../advanced_config/parameter_reference.md#COM_PREARM_MODE) to `Always`.
+
+### Configure Gripper Actuation Time
```

Comment:
I merged the profiling topic into this, but I have not re-written it yet. So consider subedit "up to here".